Wikipedia/Code/SavedArticlesFetcher.m

```diff
@@ -127,30 +128,30 @@ - (void)fetchUncachedTitles:(NSArray<MWKTitle*>*)titles {
 - (void)fetchTitle:(MWKTitle*)title {
     // NOTE: must check isCached to determine that all article data has been downloaded
     MWKArticle* articleFromDisk = [self.savedPageList.dataStore articleFromDiskWithTitle:title];
-    if (articleFromDisk.isCached) {
-        // don't fetch anything if article was cached
-        return;
-    }
-
-    /*
-       don't use "finallyOn" to remove the promise from our tracking dictionary since it has to be removed
-       immediately in order to ensure accurate progress & error reporting.
-     */
     @weakify(self);
-    self.fetchOperationsByArticleTitle[title] =
-        [self.articleFetcher fetchArticleForPageTitle:title progress:NULL].thenOn(self.accessQueue, ^(MWKArticle* article){
-        @strongify(self);
-        return [self downloadImageDataForArticle:article].thenOn(self.accessQueue, ^{
-            [self didFetchArticle:article title:title error:nil];
-        });
-    }).catch(^(NSError* error){
-        if (!self) {
-            return;
-        }
-        dispatch_async(self.accessQueue, ^{
-            [self didFetchArticle:nil title:title error:error];
+    if (articleFromDisk.isCached) {
+        // only fetch images if article was cached
+        [self downloadImageDataForArticle:articleFromDisk];
```

Comment:
@coreyfloyd 
Little tough to follow... it isCached, yet we're downloading? Does this cause images to be fetched twice? (by web view and here?)